SendDailyCampaign/SendDailyCampaign.py

```diff
@@ -10,86 +10,121 @@
+    # Scan the contacts table for a list of all contacts
+    all_contacts = scan_contacts_table()
+    # print("Contacts scanned...", all_contacts)
 
-            except Exception as e:
-                print(e)
+    # contact item example:
+    # {'Date': '2020-01-13', 'CharacterSet': 'simplified', 'Status': 'unsubscribed', 'SubscriberEmail': 'c.emilyshea@gmail.com', 'ListId': '1'}
 
-            # Replace campaign HTML placeholders with word and level
-            campaign_contents = assemble_html_content(word,level,num_level)
+    for contact in all_contacts:
 
-            # Assemble create campaign API call payload
-            payload = assemble_payload(campaign_contents,level,level_dict)
+        # Send emails to all subscribed contacts
+        if not contact['Status'] == 'unsubscribed':
+            print("Subscribed contact:", contact['SubscriberEmail'])
+            level = contact['ListId']
+            email = contact['SubscriberEmail']
+            word_index = int(contact['ListId']) - 1
+            # print("Word index:", word_index)
 
-            print(f"HSK Level {num_level} campaign payload: {payload}")
+            # Future functionality: opportunity to choose simplified or traditional word here
 
-            # SendGrid requires campaigns to be created and then sent
-            # First create the campaign and retrieve the campaign id to call the send API
-            campaign_id = create_campaign(payload)
+            word = word_list[word_index]
+            # print("Word for contact:", word)
 
-            sendgrid_response = send_campaign(campaign_id)
+            # Should these assemble and send function calls be in the handler as opposed to this function?
+            campaign_contents = assemble_html_content(level, email, word)
 
-            # Send success/error response and notification
-            if "status" in sendgrid_response and sendgrid_response["status"] == "Scheduled":
-                print(f"Campaign {sendgrid_response['id']} for HSK Level {num_level} scheduled for send successfully.")
+            try:
+                response = send_email(campaign_contents, email, level)
+            except Exception as e:
+                print(f"Error: Failed to send email - {email}, {level}.")
+                print(e)
+        else:
+            print("Unsubscribed contact:", contact['SubscriberEmail'])
+            pass
 
-            else:
-                failure_message = f"Error: Campaign for {num_level} did not schedule successfully. SendGrid API response: " + sendgrid_response
+def get_daily_word():
 
-                print (failure_message)
+    word_list = []
 
+    # Loop through HSK levels and select and save word
+    for hsk_level in range(0,6):
+        level = str(hsk_level + 1)
+        try:
+            word = select_random_word(level)
+            word_list.append(word)
         except Exception as e:
             print(e)
 
-def store_word(word,level):
+    return(word_list)
```

Comment:
In the above code if you hit an error you are just skipping that word which is good, but in the Exception handler you should add something like `word_list.append(None)` to add in a placeholder empty value for that level. Otherwise you may end up sending users words for the wrong level

Then the rest of the code needs to be careful to handle `None` values correctly.